src/sql/base/browser/ui/infoBox/infoBox.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./media/infoBox';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { status } from 'vs/base/browser/ui/aria/aria';
+import { IThemable } from 'vs/base/common/styler';
+import { Color } from 'vs/base/common/color';
+
+export interface IInfoBoxStyles {
+	informationBackground?: Color;
+	warningBackground?: Color;
+	errorBackground?: Color;
+	successBackground?: Color;
+}
+
+export type InfoBoxStyle = 'information' | 'warning' | 'error' | 'success';
+
+export interface InfoBoxOptions {
+	text: string;
+	style: InfoBoxStyle;
+}
+
+export class InfoBox extends Disposable implements IThemable {
+	private _imageElement: HTMLDivElement;
+	private _textElement: HTMLDivElement;
+	private _infoBoxElement: HTMLDivElement;
+	private _text = '';
+	private _infoBoxStyle: InfoBoxStyle = 'information';
+	private _styles: IInfoBoxStyles;
+
+	constructor(container: HTMLElement, options?: InfoBoxOptions) {
+		super();
+		this._infoBoxElement = document.createElement('div');
+		this._imageElement = document.createElement('div');
+		this._imageElement.setAttribute('role', 'image');
+		this._textElement = document.createElement('div');
+		this._textElement.classList.add('infobox-text');
+		container.appendChild(this._infoBoxElement);
+		this._infoBoxElement.appendChild(this._imageElement);
+		this._infoBoxElement.appendChild(this._textElement);
+		if (options) {
+			this.infoBoxStyle = options.style;
+			this.text = options.text;
+		}
+	}
+
+	public style(styles: IInfoBoxStyles): void {
+		this._styles = styles;
+		this.updateStyle();
+	}
+
+
+	public get infoBoxStyle(): InfoBoxStyle {
+		return this._infoBoxStyle;
+	}
+
+	public set infoBoxStyle(style: InfoBoxStyle) {
+		this._infoBoxStyle = style;
+		this._infoBoxElement.classList.remove(...this._infoBoxElement.classList);
+		this._imageElement.classList.remove(...this._imageElement.classList);
+		this._imageElement.setAttribute('aria-label', style);
+		this._infoBoxElement.classList.add('infobox-container', style);
+		this._imageElement.classList.add('infobox-image', style);
+		this.updateStyle();
+	}
+
+	public get text(): string {
+		return this._text;
+	}
+
+	public set text(text: string) {
+		if (this._text !== text) {
+			status(text);
```

Comment:
Is always using status here appropriate? What if it's a warning/error? Seems like alert might be more appropriate then

Reply:
let me add an option to control whether it should announce text changes and use alert instead.

Reply:
I'd still keep status for info/success